src/freenet/node/CHKInsertSender.java

```diff
@@ -525,7 +525,7 @@ protected void routeRequests() {
 			}
 			
 			innerRouteRequests(next, thisTag);
-			return;
+
```

Comment:
If I ever saw treacherous indentation, this is it … 

How did you test this change? This part is in the critical area of the code. If there’s any condition that could cause an infinite loop here, it could cause endlessly spinning threads.

Reply:
I have it running on my node for a little while now. Insert threads are not climbing abnormally.
Obviously, this needs to be tested by more than just me.

Reply:
Two more files. The one has this comment "// Will either chain back to routeRequests(), or call onAccepted()." right above the return.
With the return the code never loops over the counter - why is there a counter there if it is not being used?

Reply:
Unused code actually is a concern: it should be removed because it makes the code harder to understand.

What we need to preserve is existing API, though, because plugins might be using it.